File: model/src/main/java/com/itransition/fanfictionbackend/model/Reaction.java
package com.itransition.fanfictionbackend.model;

import com.itransition.fanfictionbackend.model.lookup.ReactionLookUp;
import lombok.Data;
import lombok.EqualsAndHashCode;

import javax.persistence.Entity;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

@Data
@Entity
@Table(name = "reaction")
@EqualsAndHashCode(callSuper = true)
public class Reaction extends WithLongId {

    @ManyToOne(optional = false)
    @JoinColumn(name = "reaction_lut_id")
    private ReactionLookUp value;

    @ManyToOne(optional = false)
    @JoinColumn(name = "user_id")
    private User user;

    @ManyToOne(optional = false)
    @JoinColumn(name = "fanfic_id")
    private Fanfic fanfic;
}

File: src/main/kotlin/com/itransition/fanfictionbackend/model/Reaction.kt
package com.itransition.fanfictionbackend.model

import com.itransition.fanfictionbackend.model.lookup.ReactionLookUp
import javax.persistence.Entity
import javax.persistence.JoinColumn
import javax.persistence.ManyToOne
import javax.persistence.Table

@Entity
@Table(name = "reaction")
class Reaction : WithLongId() {

    @ManyToOne(optional = false)
    @JoinColumn(name = "reaction_lut_id")
    var value: ReactionLookUp? = null

    @ManyToOne(optional = false)
    @JoinColumn(name = "user_id")
    val user: User? = null

    @ManyToOne(optional = false)
    @JoinColumn(name = "fanfic_id")
    val fanfic: Fanfic? = null
}
